Sources/TensorFlow/Layers/Dropout.swift

```diff
@@ -73,3 +73,68 @@ public struct Dropout<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
     }
   }
 }
+
+/// A gaussian noise layer
+///
+/// Gaussian Noise adds noise sampled from a normal distribution with mean zero
+public struct GaussianNoise<Scalar: TensorFlowFloatingPoint>: ParameterlessLayer {
+  @noDerivative public let standardDeviation: Tensor<Scalar>
+  
+  /// Creates a Gaussian noise layer
+  ///
+  /// - Parameter standardDeviation: Standard deviation of the Guassian distribution
+  public init(standardDeviation: Scalar) {
+    self.standardDeviation = Tensor<Scalar>(standardDeviation)
+  }
+  
+  /// Returns output obtained by adding noise
+  /// - Parameter input: The input to the layer
+  /// - Returns: The output
+  @differentiable
+  public func callAsFunction(_ input: Tensor<Scalar>) -> Tensor<Scalar> {
+    switch Context.local.learningPhase {
+    case .training:
+      let noise = Tensor<Scalar>(randomNormal: input.shape, mean: Tensor<Scalar>(0),
+                                 standardDeviation: self.standardDeviation)
+      return input + noise
+    case .inference:
+      return input
+    }
+  }
+}
+
+/// A gaussian dropout layer
+///
+/// GaussianDropout multiplies the input with the noise sampled from a normal distribution with mean 1.0.
```

Comment:
```suggestion
/// `GaussianDropout` multiplies the input with the noise sampled from a normal distribution with mean 1.0.
///
/// Because this is a regularization layer, it is only active during training time. During inference,
/// `GaussianDropout` passes through the input unmodified.
```